CommunicationLayer/CommunicationContainer.cpp

```diff
@@ -2,12 +2,9 @@
 #include <QUdpSocket>
 
 #include "../DataLayer/DataContainer.h"
-#include "CommunicationContainer.h"
-#include "CommDeviceControl/ConnectionController.h"
-#include "CommDeviceControl/CommDeviceManager.h"
```

Comment:
Along the same reasoning, I think this file is necessary as well, where it is in charge of connecting to all the devices.